console/console-init/ui/src/modules/project/ProjectPage.tsx

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React, { useState } from "react";
+import { useLocation } from "react-router";
+import { useDocumentTitle, useA11yRouteChange } from "use-patternfly";
+import {
+  PageSection,
+  PageSectionVariants,
+  Grid,
+  GridItem,
+  Page
+} from "@patternfly/react-core";
+import { Divider } from "@patternfly/react-core/dist/js/experimental";
+import { ISortBy } from "@patternfly/react-table";
+import { DELETE_ADDRESS_SPACE } from "graphql-module/queries";
+import { compareObject } from "utils";
+import { useStoreContext, types, MODAL_TYPES } from "context-state-reducer";
+import { TablePagination, IProjectCount, ProjectHeaderCard } from "components";
+import { useMutationQuery } from "hooks";
+import { IProject } from "./components";
+import {
+  getDetailForDeleteDialog,
+  getHeaderForDeleteDialog
+} from "modules/address-space";
+import { ProjectListContainer, ProjectToolbarContainer } from "./containers";
+import { initialiseFilterForProject } from "./utils";
+
+export interface ISelectSearchOption {
+  value: string;
+  isExact: boolean;
+}
+export interface IProjectFilter {
+  filterType: string;
+  names: ISelectSearchOption[];
+  namespaces: ISelectSearchOption[];
+  type?: string;
+  status?: string;
+  projectType?: string;
+}
+export default function ProjectPage() {
+  const { dispatch } = useStoreContext();
+  useDocumentTitle("Address Space List");
+  useA11yRouteChange();
+
+  const [filter, setFilter] = useState<IProjectFilter>(
+    initialiseFilterForProject()
+  );
+  const [totalProjects, setTotalProjects] = useState<number>(0);
+  const [sortDropDownValue, setSortDropdownValue] = useState<ISortBy>();
+  const location = useLocation();
+  const searchParams = new URLSearchParams(location.search);
+  const page = parseInt(searchParams.get("page") || "", 10) || 1;
+  const perPage = parseInt(searchParams.get("perPage") || "", 10) || 10;
+  const [selectedProjects, setSelectedProjects] = useState<IProject[]>([]);
+  const [isAllSelected, setIsAllSelected] = useState<boolean>(false);
+  const refetchQueries: string[] = ["all_address_spaces"];
+  const [setDeleteProjectQueryVariables] = useMutationQuery(
+    DELETE_ADDRESS_SPACE,
+    refetchQueries
+  );
+
+  const onDeleteAll = () => {
+    dispatch({
+      type: types.SHOW_MODAL,
+      modalType: MODAL_TYPES.DELETE_ADDRESS_SPACE,
+      modalProps: {
+        onConfirm: onConfirmDeleteAll,
+        selectedItems: selectedProjects.map(as => as.name),
+        option: "Delete",
+        detail: getDetailForDeleteDialog(selectedProjects),
+        header: getHeaderForDeleteDialog(selectedProjects)
+      }
+    });
+  };
+
+  const onConfirmDeleteAll = async () => {
+    if (selectedProjects && selectedProjects.length > 0) {
+      let queryVariables: Array<{ name: string; namespace: string }> = [];
+      selectedProjects.map(
```

Comment:
Map function should return an array, if there is nothing to return, use forEach instead.